sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -26,7 +27,7 @@ type globalEndpointManager struct {
 	lastUpdateTime      time.Time
 }
 
-func newGlobalEndpointManager(clientEndpoint string, pipeline azruntime.Pipeline, preferredLocations []string, refreshTimeInterval time.Duration) (*globalEndpointManager, error) {
+func newGlobalEndpointManager(clientEndpoint string, pipeline azruntime.Pipeline, preferredLocations []string, refreshTimeInterval time.Duration, enableEndpointDiscovery bool) (*globalEndpointManager, error) {
```

Comment:
nit: lets rename this, `enableEndpointDiscovery` melts the mind of anyone trying to understand